### PR TITLE
feat: unify focus styles with new standard

### PR DESCRIPTION
**Pull request by Francois-Esquire**
## Description

Unifies focus ring across CKEditor with the new focus ring state.

## Detail

[The  global variable `--ck-focus-outer-shadow`](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-theme-lark/theme/ckeditor5-ui/globals/_focus.css#L15) has been introduced to the `:root` variables to provide elements with a unified focus ring across the editor. The `--ck-inner-shadow` has been updated to the new focus ring state with inset styling.

`@zendeskgarden/react-theming` was updated to bring the latest updates with the `xs` shadow width (`$ZD-SHADOWWIDTHS-XS`).

Screenshot 2023-05-19 at 3 57 00 PM

Screenshot 2023-05-19 at 3 57 38 PM

Screenshot 2023-05-19 at 3 59 19 PM

### CODE REVIEW

**Review by geotrev — APPROVED** [review]
LGTM

**Comment by lucijanblagonic** [comment]
The implementation looks good, but I found a bug with the original implementation of the focus outline in Toolbar. Could we please adjust the focus outline size (blue shadow) from `3px` to `2px` so there is no overlap of focus over neighboring buttons.

image
image

This will make the focus outlines on the Toolbar the same as we have it on the `Tags`(`2px` instead of `3px`).

--- 

Unrelated, could we also tackle this color of the item when the dropdown is opened.
image
It should be `blue-800`.
image

**Comment by Francois-Esquire** [comment]
> The implementation looks good, but I found a bug with the original implementation of the focus outline in Toolbar. Could we please adjust the focus outline size (blue shadow) from `3px` to `2px` so there is no overlap of focus over neighboring buttons.
> 
> image image
> 
> This will make the focus outlines on the Toolbar the same as we have it on the `Tags`(`2px` instead of `3px`).
> 
> Unrelated, could we also tackle this color of the item when the dropdown is opened. image It should be `blue-800`. image

I've corrected the focus outline size for buttons and updated the "on" text color of buttons to `blue-800`. I did notice something with disabled buttons:
Screenshot 2023-05-22 at 3 26 40 PM
It's getting a focus ring when the button is disabled in the menu.

**Review by lucijanblagonic — APPROVED** [review]
LGTM!

File: src/theme/_variables.css [hunk 1]
```diff
@@ -45,8 +46,16 @@
...
   --ck-drop-shadow: var(--ck-color-shadow-drop) 0 20px 30px 0;
+  --ck-inner-shadow:
+    inset var(--ck-color-base-background) 0 0 0 $ZD-SHADOWWIDTHS-XS,
+    inset var(--ck-color-focus-border) 0 0 0 $ZD-SHADOWWIDTHS-MD;
+  --ck-focus-outer-shadow:
+    var(--ck-color-base-background) 0 0 0 $ZD-SHADOWWIDTHS-XS,
+    var(--ck-color-focus-outer-shadow) 0 0 0 $ZD-SHADOWWIDTHS-SM;
+  --ck-focus-disabled-outer-shadow:
+    var(--ck-color-base-background) 0 0 0 $ZD-SHADOWWIDTHS-XS,
+    var(--ck-color-focus-disabled-shadow) 0 0 0 $ZD-SHADOWWIDTHS-SM;
```
Comment (jzempel) [inline src/theme/_variables.css:58]: Why was the total shadow width changed from the previous 3px? The focus for toolbar buttons is now inconsistent with Garden icon buttons.

**Review by jzempel — CHANGES_REQUESTED** [review]

**Review by jzempel — CHANGES_REQUESTED** [review]
Focus for the horizontal line is missing the inner box-shadow spacer.

File: src/theme/_variables.css (hunk 1 above)
Comment (jzempel) [reply src/theme/_variables.css:58]: > so there is no overlap of focus over neighboring buttons

@lucijanblagonic the overlap here is intended by the original designs; it was deemed more important for these buttons to mimic Garden icon buttons after which they are patterned. I feel like that intent still holds.

Comment (lucijanblagonic) [reply src/theme/_variables.css:58]: @jzempel I am OK with the decision to revert if we want to aim at consistency and we don't think overlap is much of an issue.

Comment (Francois-Esquire) [reply src/theme/_variables.css]: updated in [cc62a90](https://github.com/zendeskgarden/ckeditor/pull/90/commits/cc62a901fe50cc2d3ace7e842cf66a09163b8fbb)!

**Review by jzempel — APPROVED** [review]

**Pull request merged**